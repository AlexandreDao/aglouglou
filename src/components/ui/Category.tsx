import React from 'react'
import { View, Text, ColorValue, StyleSheet } from 'react-native'
import { TEXT_COLOR } from '@/constants/colors'

interface CategoryProps {
  title: string
  backgroundColor?: ColorValue
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 14,
    paddingHorizontal: 4,
    paddingVertical: 2,
  },
  text: {
    color: TEXT_COLOR,
    fontSize: 12,
  },
})

function Category({ title, backgroundColor = 'grey' }: CategoryProps) {
  return (
    <View style={[styles.container, { backgroundColor }]}>
      <Text style={styles.text} numberOfLines={1} ellipsizeMode="tail">
        {title}
      </Text>
    </View>
  )
}

export default Category
