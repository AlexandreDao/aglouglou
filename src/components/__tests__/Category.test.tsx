import React from 'react'
import { render } from '@testing-library/react-native'
import Category from '@/components/ui/Category'

describe('<Category />', () => {
  test('Category snapshot', () => {
    const tree = render(<Category title="Hello world" />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
