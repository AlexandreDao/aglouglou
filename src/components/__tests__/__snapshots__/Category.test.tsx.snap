// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> Category snapshot 1`] = `
<View
  style={
    [
      {
        "borderRadius": 14,
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      },
      {
        "backgroundColor": "grey",
      },
    ]
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "color": "#EDEEF0",
          "fontSize": 12,
        },
        undefined,
      ]
    }
  >
    Hello world
  </Text>
</View>
`;
